x-pack/plugins/siem/server/lib/exceptions/task.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createHash } from 'crypto';
+import { CoreSetup, Logger, SavedObjectsClient } from '../../../../../../src/core/server';
+import {
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '../../../../../plugins/task_manager/server';
+import { ListPluginSetup } from '../../../../lists/server';
+import { GetFullEndpointExceptionList, CompressExceptionList } from './fetch_endpoint_exceptions';
+
+export interface PackagerTask {
+  getTaskRunner: (context: PackagerTaskRunnerContext) => PackagerTaskRunner;
+}
+
+interface PackagerTaskRunner {
+  run: () => void;
+}
+
+interface PackagerTaskContext {
+  core: CoreSetup;
+  logger: Logger;
+  taskManager: TaskManagerSetupContract;
+  lists: ListPluginSetup;
+}
+
+interface PackagerTaskRunnerContext {
+  taskManager: TaskManagerStartContract;
+}
+
+export function setupPackagerTask(context: PackagerTaskContext): PackagerTask {
+  const run = async () => {
+    context.logger.debug('Running exception list packager task');
+
+    const [{ savedObjects }] = await context.core.getStartServices();
+    const savedObjectsRepository = savedObjects.createInternalRepository();
+    const soClient = new SavedObjectsClient(savedObjectsRepository);
+    const exceptionListClient = context.lists.getExceptionListClient(soClient, 'kibana');
+    const exceptions = await GetFullEndpointExceptionList(exceptionListClient);
+    const compressedExceptions: Buffer = await CompressExceptionList(exceptions);
+
+    const sha256Hash = createHash('sha256')
+      .update(compressedExceptions.toString('utf8'), 'utf8')
+      .digest('hex');
+
+    const exceptionSO = {
+      name: 'global-whitelist',
+      schemaVersion: '1.0.0',
+      sha256: sha256Hash,
+      encoding: 'xz',
+      created: Date.now(),
+      body: compressedExceptions.toString(),
+    };
+
+    const resp = await soClient.find({
+      type: 'siem-exceptions-artifact',
+      search: sha256Hash,
+      searchFields: ['sha256'],
+      fields: [],
+    });
+
+    // TODO clean this up and handle errors better
```

Comment:
To clarify the TODO: This is currently not atomic (just hooked up quickly for purposes of demo). We should plan to make this atomic, if possible. Do saved objects let us pass in our own `_id`? If so, we can just do an optimistic write here. If not possible to make this atomic, we'll need to do some dupe detection on the retrieval.